src/map/entities/charentity.h

```diff
@@ -212,6 +213,8 @@ class CCharEntity : public CBattleEntity
     UnlockedAttachments_t	m_unlockedAttachments;			// Unlocked Automaton Attachments (1 bit per attachment)
     CAutomatonEntity*       PAutomaton;                     // Automaton statistics
 
+    CLuopanEntity* PLuopan;
```

Comment:
I don't remember if this is even used anymore, I think we discovered that the Luopan takes up the pet slot, so it can't be used with other pets (mobs, jugs, fellows, wyverns etc.). Probably best to check that though

Reply:
It says you already have a pet if you try to cast an avatar, and very obviously uses the same placement for the loupans HP bar, so yeah.

Reply:
Oh i missed this one

Reply:
done